modules/contact-form/grunion-contact-form.php

```diff
@@ -2554,6 +2554,23 @@ function process_submission() {
 			self::wp_mail( $to, "{$spam}{$subject}", $message, $headers );
 		}
 
+		/**
+		 * Fires an action hook right after the email(s) have been sent.
+		 *
+		 * @module contact-form
+		 *
+		 * @since 6.6.2
+		 *
+		 * @param int $post_id Post contact form lives on.
+		 * @param string|array $to Array of valid email addresses, or single email address.
+		 * @param string $subject Feedback email subject.
+		 * @param string $message Feedback email message.
+		 * @param string|array $headers Optional. Additional headers.
+		 * @param array $all_values Contact form fields.
+		 * @param array $extra_values Contact form fields not included in $all_values
+		 */
+		do_action( 'grunion_after_message_sent', $post_id, $to, $subject, $message, $headers, $all_values, $extra_values );
```

Comment:
That's a lotta parameters.  Part of me wonders if we should `compact()` to, subject, message, and headers into one associative array.

Reply:
I will work on this change and update the PR tonight.

Reply:
As I was looking for the best way to reduce the number of parameters, as simple `compact()` style of reduction would result in the possible array of `$to` items and the array of `$headers` all being jammed into a single dimension array. Which would also result in a failed/problematic `extract()`. So, while yes it is a seemingly large set of parameters is the best one to give the most flexibility, and provide the most options for post-processing.

Reply:
For a bit of prior art, https://developer.wordpress.org/reference/hooks/wp_mail/ compacts those items for the filter, but WP doesn't have a post-send action on `wp_mail` for anything to mimic. There is another hook with a number of parameters too ( https://core.trac.wordpress.org/browser/tags/5.1/src/wp-includes/pluggable.php#L915 ).

I'm fine letting this go in uncompacted.